lib/msprime.c

```diff
@@ -5893,6 +5893,24 @@ msp_activate_population(msp_t *self, int population_id)
     return ret;
 }
 
+/* Adds a population activation event.
+ * Note: we use the int32_t type here so we can be sure we're passing in
+ * arrays of the correct type from numpy where we have to specify the size.
+ */
+int MSP_WARN_UNUSED
+msp_add_activate_population(
```

Comment:
This function needs to create a demographic event for the population activation (e.g. see the `msp_add_population_split()` function). The demographic event's `change_state` field should be a pointer to the function that does the activation (you'll need to write this function, which I guess will be a thin wrapper around `msp_activate_population`). The "activate population" demographic event's `change_state()` function will then be triggered at the appropriate time from the simulation's main loop (see `msp_run_coalescent()`, which calls `msp_apply_fixed_events()`, etc.). You'll need to add a struct in msprime.h for the demographic event (eg. see the one for the struct typedef'ed to `population_split_t`).